register/src/register.rs

```diff
@@ -0,0 +1,366 @@
+                .arg(
+                    Arg::with_name("details")
+                        .short("d")
+                        .long("details")
+                        .value_name("STRING")
+                        .takes_value(true)
+                        .validator(check_details_length)
+                        .help("Validator description, max characters: ###"),
+                ),
+        )
+        .subcommand(
+            SubCommand::with_name("update")
+                .about("Update Validator registration info")
+                .setting(AppSettings::DisableVersion)
+                .arg(
+                    Arg::with_name("name")
+                        .short("n")
+                        .long("name")
+                        .value_name("STRING")
+                        .takes_value(true)
+                        .required(true)
+                        .validator(check_name_length)
+                        .help("Validator name"),
+                )
+                .arg(
+                    Arg::with_name("website")
+                        .short("w")
+                        .long("website")
+                        .value_name("URL")
+                        .takes_value(true)
+                        .validator(check_url)
+                        .help("Validator website url"),
+                )
+                .arg(
+                    Arg::with_name("keybase")
+                        .short("k")
+                        .long("keybase-id")
+                        .value_name("STRING")
+                        .takes_value(true)
+                        .validator(is_keybase)
+                        .help("Validator Keybase id"),
+                )
+                .arg(
+                    Arg::with_name("details")
+                        .short("d")
+                        .long("details")
+                        .value_name("STRING")
+                        .takes_value(true)
+                        .validator(check_details_length)
+                        .help("Validator description, max characters: ###"),
+                ),
+        )
+        .get_matches();
+
+    let json_rpc_url = matches.value_of("json_rpc_url").unwrap();
+
+    let mut path = dirs::home_dir().expect("home directory");
+    let id_path = if matches.is_present("validator_keypair") {
+        matches.value_of("validator_keypair").unwrap()
+    } else {
+        path.extend(&[".config", "solana", "validator-keypair.json"]);
+        if !path.exists() {
+            println!("No validator keypair file found. Run solana-keygen to create one.");
+            exit(1);
+        }
+        path.to_str().unwrap()
+    };
+    let validator_keypair = read_keypair(id_path)?;
+
+    let mut path = dirs::home_dir().expect("home directory");
+    let id_path = if matches.is_present("info_keypair") {
+        matches.value_of("info_keypair").unwrap()
+    } else {
+        path.extend(&[".config", "solana", "validator-info.json"]);
```

Comment:
Who creates `~/.config/solana/validator-info.json`?

Reply:
Currently, "Validators are responsible for generating a validator-info keypair and passing it to the CLI as the registration info account location and signer." (from PR description)
We could have the validator-info script generate that keypair, but that seems awkward from a real-world key-management perspective.

Reply:
Got it.  Can you add the extra steps to the testnet participation guide

Reply:
Will do!